shared/src/test/kotlin/com/egm/stellio/shared/util/JsonLdUtilsTests.kt

```diff
@@ -95,4 +98,42 @@ class JsonLdUtilsTests {
         Assertions.assertTrue(resultMap.containsKey("location"))
         Assertions.assertFalse(resultMap.containsKey("isParked"))
     }
+
+    @Test
+    fun `it should throw an InvalidRequest exception if the JSON-LD fragment is not a valid JSON document`() {
+        val rawEntity =
+            """
+            {
+                "id": "urn:ngsi-ld:Device:01234",,
+                "type": "Device"
+            }
+            """.trimIndent()
+
+        val exception = assertThrows<InvalidRequestException> {
+            parseAndExpandJsonLdFragment(rawEntity)
+        }
+        Assertions.assertNotNull(exception.message)
+    }
+
+    @Test
+    fun `it should throw a BadRequestData exception if the JSON-LD fragment is not a valid JSON-LD document`() {
+        val rawEntity =
+            """
+            {
+                "id": "urn:ngsi-ld:Device:01234",
+                "type": "Device",
+                "@context": [
+                    "unknownContext"
+                ]
+            }
+            """.trimIndent()
+
+        val exception = assertThrows<BadRequestDataException> {
+            parseAndExpandJsonLdFragment(rawEntity)
+        }
+        Assertions.assertEquals(
+            "Unexpected error while parsing payload : loading remote context failed: unknownContext",
+            exception.message
+        )
+    }
```

Comment:
no way to also the third error case in `parseAndExpandJsonLdFragment`?

Reply:
I added a test for the third case